### ISSUE TITLE
Knife Juggler, Imp Gang Boss, and Grim Patron Interaction.

### ISSUE THREAD

**Issue by ikhaliq15** [demilich1/metastone#342]
So according to the Advanced rulebook, the follow interaction should leave you with no grim patron.
![screen shot 2017-04-19 at 12 01 17 am](https://cloud.githubusercontent.com/assets/6558567/25192295/86ad1a8a-24e7-11e7-8c52-0e22c67ec7ff.png)

However, when I ran it in the Metastone game, I was left with a grim patron remaining. I did everything according to how it says and you can watch a video of the whole interaction [here](https://www.youtube.com/watch?v=rlEQHNy23k0).

I start with:
![screen shot 2017-04-19 at 12 35 19 am](https://cloud.githubusercontent.com/assets/6558567/25192376/d1f92218-24e7-11e7-87ce-6d0a1c697585.png)

and end with:

![screen shot 2017-04-19 at 7 23 19 am](https://cloud.githubusercontent.com/assets/6558567/25192207/2f3265f8-24e7-11e7-9073-3fda8ab52792.png)

I don't know if this interaction is to specific to care about, but just thought I would write about it here.

**Comment by webadict** [comment]
You are correct. I actually saw this a long time ago, but never got around to fixing it. I know exactly what needs to be changed for this, but it's a bit of work, so I'll probably push it a while back.

**Comment by doctorpangloss** [comment]
@webadict described the issue accurately and this is now fixed in my fork.
